### PR TITLE
Do not run GPU tests and Docker workflows on release.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Some test workflows are run on release (GPU ones) and not others.
The Docker build workflow is also run, but it it missing required inputs.

**Description of the Change:**
Do not run the mentioned workflows on release. The Docker workflow in particular cannot be run because it requires the new PennyLane release which is not available when Lightning is released. 

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-68101]

### CODE REVIEW

File: .github/workflows/compat-docker-latest.yml [hunk 1]
```diff
@@ -1,20 +1,20 @@
-name: Compat Check Docker - Lightning@latest
+name: Compat Check Docker - Lightning@stable
```
Comment (AmintorDusko) [inline .github/workflows/compat-docker-latest.yml:1]: Stable because, our latest already diverged from stable, right?

Comment (AmintorDusko) [reply .github/workflows/compat-docker-latest.yml:1]: Should we rename the script to stable? Or do we use it for latest too?

Comment (vincentmr) [reply .github/workflows/compat-docker-latest.yml:1]: I realized that we refer to the latest official release as "stable" and the development (also our master branch) as "latest". I'm thus adopting the convention in the Docker workflow.

Comment (AmintorDusko) [reply .github/workflows/compat-docker-latest.yml:1]: In this case, should we rename the script to:
.github/workflows/compat-docker-stable.yml ?

Comment (vincentmr) [reply .github/workflows/compat-docker-latest.yml:1]: Good point. Will do.

File: .github/workflows/compat-docker-latest.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     name: Docker latest - Linux::x86_64
     uses: ./.github/workflows/docker_linux_x86_64.yml
     with:
-      lightning-version: v0.36.0
-      pennylane-version: v0.36.0
+      lightning-version: master
+      pennylane-version: master
```
Comment (AmintorDusko) [inline .github/workflows/compat-docker-latest.yml:18]: Isn't the latest tag applicable here?

File: .github/workflows/compat-docker-stable.yml [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-  docker_linux_x86_64_develop:
-    name: Docker develop - Linux::x86_64
+  docker_linux_x86_64_stable:
+    name: Docker stable - Linux::x86_64
     uses: ./.github/workflows/docker_linux_x86_64.yml
     with:
-      lightning-version: master
-      pennylane-version: master
+      lightning-version: v0.37.0
+      pennylane-version: v0.37.0
```
Comment (AmintorDusko) [inline .github/workflows/compat-docker-stable.yml:18]: Isn't the stable tag applicable here?

**Review by AmintorDusko — COMMENTED** [review]
some more questions.

File: .github/workflows/compat-docker-stable.yml (hunk 3 above)
Comment (vincentmr) [reply .github/workflows/compat-docker-stable.yml:18]: No, we do not maintain a `stable` tag. We could use `latest_release` for `lightning-version` but PennyLane does not have an equivalent tag it appears.

**Review by AmintorDusko — APPROVED** [review]
I have no blockers, but we could propose creating the necessary tags.
Having these tags in place would be a more maintainable solution.

**Comment by vincentmr** [comment]
> I have no blockers, but we could propose creating the necessary tags. Having these tags in place would be a more maintainable solution.

Agreed. Let's do in a follow-up PR? I believe we'll soon improve on everything while including Catalyst into the mix, which will further motivate creating a more maintainable solution. We can then see with core and compilation what tags they would all like to use.

**Comment by AmintorDusko** [comment]
> > I have no blockers, but we could propose creating the necessary tags. Having these tags in place would be a more maintainable solution.
> 
> Agreed. Let's do in a follow-up PR? I believe we'll soon improve on everything while including Catalyst into the mix, which will further motivate creating a more maintainable solution. We can then see with core and compilation what tags they would all like to use.

Agreed!

**Pull request merged**